ONVIF device triggers and homeassistant event

teaserrr opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
This PR is based largely on a [PR from 2020](https://github.com/home-assistant/core/pull/36428) by @hunterjm. All changes are his, I merely fixed the things needed because of the evolution of the HA codebase. Note that this is my first contribution to HA, hopefully everything is according to the guidelines.

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This pull request allows us to handle firing home assistant events for ONVIF events that can not be represented via entities as they have no state. To make it easier to use in automations, it also implements device triggers for these events.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/13660 (closed)

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

TimSlechten committed:
add ability to trigger ha events

TimSlechten committed:
black/mypy fixes

teaserrr commented:
Maybe this should be reviewed by someone else since @hunterjm was the actual author of these changes. Maybe @MartinHjelmare who reviewed the original PR?

emontnemery commented on homeassistant/components/onvif/device_trigger.py:
```diff
@@ -0,0 +1,78 @@
+            continue
+
+        device: ONVIFDevice = hass.data[DOMAIN][config_entry.unique_id]
+        for event in device.events.get_platform("event"):
```
As I understand it, the possible onvif events defined in `homeassistant/components/onvif/parsers.py`?
If so, the events are currently using a name which is English natural language.

For the device triggers, it would be better to map non natural language to natural language, for example: `motion_alarm` ->`Motion Alarm`

Maybe add a new key `event_id` to `homeassistant.components.onvif.models.Event`?

Note: Translatable entity names will come soon: <https://github.com/home-assistant/core/pull/88242>, the entities generated from events should then have their names translated.

teaserrr replied:
I wasn't sure this was relevant as the event name is only used in the subtype of the event, nonetheless I implemented a conversion to non-natural language.

teaserrr committed:
use non-natural language key in CONF_SUBTYPE

teaserrr committed:
Add event_id attribute for onvif events

teaserrr commented on homeassistant/components/onvif/models.py:
```diff
@@ -78,3 +80,6 @@ class Event:
     enabled: bool = True
+
+    def get_event_id(self) -> str:
+        return re.sub('\W+', '_', self.name.lower())
```
I did the requested changes but one of the event ids is created dynamically. I checked the documentation you linked to but couldn't find an example of such a case. Are translations like this supported: `%s_line_crossed` -> `%s Line Crossed` ?

teaserrr replied:
@emontnemery can you have a look at the changes please?

bdraco committed:
Significantly improve performance of conversation default_agent listening for new states (#93577)

Use the async_track_state_added_domain helper instead of tracking
all state changes and rejecting them as it is already optimized
for this job

bdraco committed:
Complete persistent notifications migration (#92828)

* Complete migration of persistent notifications

Persistent notifications are no longer stored in
the state machine and no longer fire events

* Complete migration of persistent notifications

Persistent notifications are no longer stored in
the state machine and no longer fire events

* fixes

* fixes

* fixes

* ws test

* update tests

* update tests

* fix more tests

* fix more tests

* more fixes

* fix

* fix person

* fix person

* keep whitelist

* use singleton

bdraco committed:
Increase timeout for test_purge_many_old_events (#93579)

frenck committed:
Add entity name translations to Rituals Perfume Genie (#92549)

jbouwh committed:
Add type hints for stub_blueprint_populate fixture (#93568)

Add typehints for stub_blueprint_populate fixture

jbouwh committed:
Remove redundant stub_blueprint_populate fixture for MQTT integration (#93570)

Remove redundant stub_blueprint_populate fixture

gjohansson-ST committed:
Remove platform yaml Frontier Silicon (#93552)

gjohansson-ST committed:
Remove EDL21 YAML configuration (#93551)

gjohansson-ST committed:
Remove obihai YAMl configuration (#93549)

bramkragten committed:
Update frontend to 20230526.0 (#93581)

emontnemery committed:
Fix mill tests (#93596)

emontnemery committed:
Fix sharkiq tests (#93597)

pail23 committed:
Use entity attributes in mystrom (#93591)

* Migrated proprties to _attr_

* Update homeassistant/components/mystrom/light.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/light.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/light.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/light.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/light.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/light.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/switch.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/mystrom/switch.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* review comment

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

emontnemery committed:
Add CONFIG_SCHEMA to analytics (#93598)

rajlaud committed:
Bump pysqueezebox to 0.6.2 (#93573)

Drafteed committed:
Bump androidtvremote2 to 0.0.9 (#93634)

boralyl committed:
Fix todoist end time for tasks with due date in the future (#91874)

Fix end time for tasks with due date in the future.

Co-authored-by: Allen Porter <allen@thebends.org>

joostlek committed:
Add YouTube integration (#92988)

* Add YouTube stub

* Add YouTube stub

* Add YouTube stub

* Add YouTube stub

* Add Youtube stub

* Add Youtube stub

* Add tests

* Add tests

* Add tests

* Clean up

* Add test for options flow

* Fix feedback

* Fix feedback

* Remove obsolete request

* Catch exceptions

* Parallelize latest video calls

* Apply suggestions from code review

Co-authored-by: Robert Hillis <tkdrob4390@yahoo.com>

* Add youtube to google brands

* Fix feedback

* Fix feedback

* Fix test

* Fix test

* Add unit test for http error

* Update homeassistant/components/youtube/coordinator.py

Co-authored-by: Robert Hillis <tkdrob4390@yahoo.com>

* Fix black

* Fix feedback

* Fix feedback

* Fix tests

---------

Co-authored-by: Robert Hillis <tkdrob4390@yahoo.com>

Noltari committed:
Update aioairzone to v0.6.1 (#93629)

bdraco committed:
Switch some frequently called call_laters to call_at (#93659)

* Switch some frequently called call_laters to call_at

call_at is a bit faster than call_later since call_later
is a wrapper around call_at.

We call at lot of these at startup so it helps a bit when
we are resource constrained

* update test

* update test

bdraco committed:
Use httpx instead of requests for mjpeg camera images that need digest to avoid jump to executor (#93244)

* Use httpx instead of requests for mjpeg camera images that need digest

Avoids jump to executor

* Use httpx instead of requests for mjpeg camera images that need digest

Avoids jump to executor

* stream as well

* fix

* fix

bdraco committed:
Reduce overhead to filter entities in the recorder (#93656)

* Reduce overhead to filter entities in the recorder

* fix type

bdraco committed:
Collapse valid state check when creating a state (#93655)

Inline valid state check when creating a state

This was added in #9696 and only used in one place so
it does not need to be another function

bdraco committed:
Improve performance when serializing small bits of JSON (#93653)

* Improve performance when serializing small bits of JSON

Making json_bytes a partial reduced the run time to
build the small JSON messages by ~18.75%

We serialize a lot of small messages over the websocket

* typing

bdraco committed:
Disable polling on energy sensors (#93651)

These sensors do not implement async_update so no
polling is needed

bdraco committed:
Disable polling for samsungtv remote entities (#93650)

These entities do not have a defined state, there is no reason to poll

bdraco committed:
Await in sequence when we know we will be blocked by the entity platform semaphore (#93649)

* Defer creating tasks we know are going to wait on the entity platform semaphore

When looking at whats going on with aiomonitor-ng, I noticed
we end up creating a lot of tasks that block waiting for the
executor because of the entity platform parallel_updates semaphore.

When we know the tasks are going to block we now await them
in sequence to avoid feeding the loop a herd of tasks that
will block on the semaphore

* change during iteration fix

* change during iteration fix

* cleanup

* cleanup

* fix vizio test

bdraco committed:
Disable call_soon for device tracker device registry updates (#93638)

Run device tracker device registry updates immediately
without the call_soon delay. This listener does not
need a delay and it only adds overhead in this case.

bdraco committed:
Speed up processing subscribe_events and subscribe_entites when user has read all permissions (#93611)

Speed up processing subscribe_events and subscribe_entites when user the read all permissions

bdraco committed:
Add event helper to dispatch device registry updates by device_id (#93602)

* Add event helper to dispatch device registry updates by device_id

* Update homeassistant/helpers/event.py

* dry

* dry

* reduce

* reduce

* reorder

* reduce

* cleanup stack depth

* dry

* fix double lookup

* remove unused

* collapse

bdraco committed:
Make diagnostics dataclasses sloted (#93600)

Now available since our minimum python version is 3.10

Saves a tiny bit of ram

joostlek committed:
Add reauth flow to Youtube (#93670)

bdraco committed:
Add hardware model to onvif config flow discovery (#93676)

joostlek committed:
Improve OpenSky typing (#93666)

* Add types to OpenSky

* Update .strict-typing

* Add types to OpenSky

emontnemery committed:
Use config entry setup in cast tests (#93595)

* Use config entry setup in cast tests

* Remove import step from config flow

* Remove import tests

* Fix tests

wrt54g committed:
Update dialogflow URL (#93616)

Update URL

joostlek committed:
Improve Youtube integration (#93683)

* Fix feedback

* Fix feedback

jbouwh committed:
Fix imap parsing non rfc compliant date crash (#93630)

* Fix imap parsing non rfc compliant date crash

* Use parsedate_to_datetime from mail.utils

farmio committed:
Update xknx to 2.10.0 and knx-frontend to 2023.5.28.94855 (#93694)

emontnemery committed:
Make cv.empty_config_schema log an error instead of raise (#93646)

* Make cv.empty_config_schema log an error instead of raise

* Add test

* Update homeassistant/helpers/config_validation.py

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

bdraco committed:
Fix refactoring error with updating polling entities in sequence (#93693)

* Fix refactoring error with updating in sequence

see #93649

* coverage

* make sure entities are being updated in parallel

* make sure entities are being updated in sequence

emontnemery committed:
Fix typo in ping (#93701)

joostlek committed:
Improve async youtube (#93685)

* Improve async youtube

* Improve async youtube

gjohansson-ST committed:
Raise issue "does not support platform setup" (#93585)

* Raise issue on platform missing

* Modify issue

* Remove deprecation

* Fix strings

* Strings

* Last strings

* strings to common

emontnemery committed:
Fix typo in switcher_kis (#93702)

marvin-w committed:
Restore original websocket commands for config entries (#93707)

Restore original websocket commands and add "config_entries/get_single"

joostlek committed:
Add Options flow to YouTube (#93667)

* Add Options flow to YouTube

* Add strings for options flow

* Add strings for options flow

* Add strings for options flow

gjohansson-ST committed:
Move yaml configuration to integration key for command_line (#92824)

* Inital init commit

* bs and cover

* notify

* sensor

* switch

* Issues

* Finalize __init__

* First pass tests

* Fix Binary sensors

* Test cover

* Test notify

* Test sensor

* Tests switch

* Fix coverage

* Add codeowner

* Fix caplog

* test issue

* Flaky test notify

* Fix async

* Adjust yaml structure

* Change yaml format again

* Issue strings

* Fix tests

* string review comment

jbouwh committed:
Do not trigger reload when unloading or reloading MQTT entry (#93588)

* Do not trigger reload when unloading MQTT entry

* More cleanup

* cleanup async_reload_manual_mqtt_items

* Add test

* Improve test

* Remove unuse mocks from test

* Add discovery item in test

bdraco committed:
Switch mqtt to use async_track_device_registry_updated_event (#93603)

bdraco committed:
Fix thread leak in onvif discovery (#93742)

mkmer committed:
Bump Blinkpy to 0.21.0 (#93735)

bdraco committed:
Drop codeowner for radiotherm (#93744)

Lash-L committed:
Add Roborock Clean time sensors (#93637)

* add clean time information

* add sensor test

* update to cleaning time

* fix sensor test

StephanU committed:
Bump pysml to 0.0.12 (#93686)

gjohansson-ST committed:
Address late review comments for Command Line (#93743)

* Late review comments

* Rephrase

elupus committed:
Fix missed case for device tuple fallback in rfxtrx (#93575)

* rfxtrx: add missed case for device tuple fallback

Previously the code was prepared for a backward compatible
migration of tuples, where both styles of identifiers might exist
in device registry at the same time.

This place was sadly missed

* No need to raise special, we can assert instead

* Add some basic tests

elupus committed:
Don't mock non async with AsyncMock (#93749)

Noltari committed:
Fix possible KeyError exception in Airzone Cloud (#93747)

airzone_cloud: fix KeyError for Aidoo installations

Some Airzone Cloud installations might be limited to Aidoo devices. In that
case, the library will provide AZD_AIDOOS, but it won't provide AZD_ZONES,
resulting in a KeyError exception.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

scop committed:
Access datetime.datetime and .time directly in caldav (#93688)

gjohansson-ST committed:
Remove nextcloud YAML configuration (#93550)

emontnemery committed:
Add empty config schema to integrations a-c (#93608)

bdraco committed:
Reduce overhead to update unifiprotect entity (#93725)

yuxincs committed:
Fix unit inference for ITEMP field for APCUPSD integration (#93724)

bdraco committed:
Handle exceptions to prevent disconnect when esphome state subscription raises (#93723)

jgarec committed:
Add overkiz control for ATLANTIC_PASS_APC_HEATING_ZONE (#93712)

bdraco committed:
Make unit converter use a factory to avoid looking up the ratios each conversion (#93706)

bdraco committed:
Fix onvif setup failing when unable to parse camera time (#93677)

emontnemery committed:
Add empty config schema to integrations p-s (#93700)

emontnemery committed:
Add empty config schema to integrations i-o (#93698)

emontnemery committed:
Add empty config schema to integrations d-h (#93645)

emontnemery committed:
Detect attempt to setup config entry integration via YAML (#93589)

Noltari committed:
Add Aidoo sensors to Airzone Cloud (#93541)

scop committed:
Import `util.dt` as `dt_util` in `components/[e-f]*` (#93758)

scop committed:
Import `util.dt` as `dt_util` in `components/[t-z]*` (#93763)

scop committed:
Import `util.dt` as `dt_util` in `components/[g-i]*` (#93759)

scop committed:
Import `util.dt` as `dt_util` in `components/[a-d]*` (#93756)

scop committed:
Import `util.dt` as `dt_util` in `components/[k-o]*` (#93760)

scop committed:
Import `util.dt` as `dt_util` in `components/[p-s]*` (#93761)

raman325 committed:
Add `datetime` platform (#81943)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>

rikroe committed:
Ensure state update after BMW remote service execution (#93745)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

joostlek committed:
Add entity name translations to Buienradar (#91511)

Co-authored-by: Erik Montnemery <erik@montnemery.com>

bdraco committed:
Reduce overhead to match homekit zeroconf discoveries (#93753)

mover85 committed:
Bump pydaikin 2.9.1 (#93635)

rikroe committed:
Add charging switch to BMW Connected Drive (#93737)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

bdraco committed:
Reduce overhead to fetch integrations (#93767)

We call this path over and over during startup and most
of the time the integration is already loaded. We want
that case to be the short path

bdraco committed:
Avoid dict comp in statistic_during_period if there is no conversion (#93699)

bdraco committed:
Add a filter to the tasmota device registry listener (#93640)

We can avoid creating a task when the event is not
a remove which will be most cases

cdce8p committed:
Update pydantic to 1.10.8 (#93772)

allenporter committed:
Fix failing todoist test that used timezone before set by fixture (#93775)

* Fix failing todoist test that used timezone before set by fixture

* Merge fix for dt -> dt_util

Lash-L committed:
Remove unneeded icon for Roborock sensor (#93774)

removed unneeded icon

bdraco committed:
Speed up creating automations (#93776)

Creating the inner coroutine for each automation to be run at started
was expensive when the user had 1000s of automations

gjohansson-ST committed:
Use entity attributes in Obihai sensor (#93564)

emontnemery committed:
Adapt otbr to upstream openthread REST API changes (#93544)

* Adapt otbr to upstream openthread REST API changes

* Bump python-otbr-api to 2.0.0

Noltari committed:
Add WebServer sensors to Airzone Cloud (#93757)

bdr99 committed:
Fix broken URL in MQTT discovery log message (#93778)

frenck committed:
Merge branch 'master' into dev

emontnemery committed:
Bump python-otbr-api to 2.1.0 (#93790)

* Bump python-otbr-api to 2.1.0

* Fix tests

Diegorro98 committed:
Add tilt support to Matter cover (#92256)

frenck committed:
Delay release of HA Python release until Docker is ready (#93798)

emontnemery committed:
Teach search about blueprints (#78535)

emontnemery committed:
Deduplicate Thread datasets based on EXTPANID (#93795)

* Deduplicate Thread datasets based on EXTPANID

* Apply suggestions from code review

PeteRager committed:
Sonos setup fails with unhandled exceptions on discovery messages (#90648)

Co-authored-by: J. Nick Koston <nick@koston.org>

raman325 committed:
Add support for Window Covering CC (#93314)

* Refactor zwave_js.cover and improve test coverage

* Remove extra fixtures

* cleanup old stuff

* Get coverage to 100

* Add support for Window Covering CC

* fix bug

* typo

* remove redundant stuff

* slight change to improve readability

* Add device class

* Update to match specs

emontnemery committed:
Fix race in rfxtrx config flow (#93804)

* Fix race in rfxtrx config flow

* Add timeout

* Use async_timeout.timeout

raman325 committed:
Add zwave config parameter entities (#92223)

* Add zwave config parameter entities

* Remove unused entity const

* remove unusued imports

* review comments

* switch to reserved values

* fix test

frenck committed:
Deprecate Python 3.10 (#93794)

Lash-L committed:
Add more information to Roborock Status (#93636)

Co-authored-by: Humberto Gontijo <humberto.gontijo@clevertech.biz>

Lash-L committed:
Add translations to Roborock fan speed (#93801)

jbouwh committed:
Fix imap advanced settings not shown (#93813)

gjohansson-ST committed:
Fix switch acting as toggle in Sensibo (#93810)

* Fix switch as toogle

* typing

bdraco committed:
Remove outdated HAOS check from bluetooth (#93809)

bdraco committed:
Fix double state write on camera added to hass when rtsp_to_webrtc is present (#93806)

frenck committed:
Use deployment environment in GitHub build actions (#93800)

jbouwh committed:
Add ssl_verify option to imap integration (#93811)

* Add ssl_verify option to imap integration

* Add test

raman325 committed:
Add zwave_js node statistics sensors (#91714)

* Add node statistics sensors

* fix tests and don't let controller state leak across tests

* Add background RSSI

* Remove extra logging statement

* fix test

* comments

* setup platform once

* Add static properties to entity description

* Update homeassistant/components/zwave_js/sensor.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* don't dupe attribute values in entity description

* fix exception

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

mdegat01 committed:
Add docker config repair for supervisor issue (#93820)

synesthesiam committed:
Bump intents to 2023.5.30 (#93822)

synesthesiam committed:
Don't skip chunk before speech (#93825)

scop committed:
Import `util.dt` as `dt_util` in remaining tests, enforce with ruff (#93765)

* Import `util.dt` as `dt_util` in remaining tests, enforce with ruff

* Fix `util.dt` import in pvpc_hourly_pricing tests

frenck committed:
Get nightly intents from new intents-package repo (#93826)

Get nighlty intents from new intents-package repo

starkillerOG committed:
Fix false reolink ONVIF webhook warning (#91015)

jesserockz committed:
Add speech detection and conversation_id to esphome voice assistant (#93578)

* Add speech detection to esphome voice assistant

* Timeout after silence
Ensure events are sent before finish is called

* use va_version 3 instead of ESPHome version

* Convert repeated fixtures to factory

* Add some v3 tests

* Add conversation_id

* Bump aioesphomeapi to 13.8.0

* Fix missed buffering of detected chunk

* Alter log message

* Updates

* Spelling

* Fix return type

tetele committed:
Fix error if a `custom_sentences` file is empty (#93530)

* Fix #93528

* Log warning file is invalid

* More explicit log warning message

* Rewrite log message

yuxincs committed:
Directly check string representation of sensor states in APCUPSD tests (#93783)

* Directly check the string representation of sensor states

* Fix expected state value for sensor.ups_nominal_output_power

agners committed:
Update iotawattpy to use our own fork (#93830)

Unfortunately iotawattpy got dormant. Use our fork hosted at in the
[home-assistant-libs/iotawattpy](https://github.com/home-assistant-libs/iotawattpy)
repository and available on PyPI as ha-iotawatt instead.

The first release of the fork also contains a fix for issue #82869.

bdraco committed:
Use ReadOnlyDict for entity registry options (#93824)

* Use ReadOnlyDict for entity registry options

While reviewing #93601 it was noticed this was slow at startup
https://github.com/home-assistant/core/pull/93601#issuecomment-1568958280

This is a first pass attempt to improve the performance

* fix tests

bdraco committed:
Speed up setting up event trackers (#93823)

noticed in https://github.com/home-assistant/core/pull/93601 that
the cost of creating the function in the closure was a bit expensive
since we do it once per entity

andarotajo committed:
Add coordinator to dwd_weather_warnings (#93720)

* Add data coordinator

* Specify coordinator data type

* Apply code review changes

jbouwh committed:
Remove mqtt sensor support for `last_reset_topic` (#93792)

* Remove mqtt sensor support for `last_reset_topic`

* Add cases for empty last reset messages

Lash-L committed:
Add Switch platform to Roborock (#93833)

* init

* add switch platform

* remove stale comments

* remove stale list

* set entity category to config

bdraco committed:
Add a debounce to ld2410 to prevent it from overwhelming the state machine with many devices (#93819)

* Add a debounce to ld2410 to prevent it from overwhelming the state machine

fixes #86665

* set time on debounce

gjohansson-ST committed:
Translate entity names for Yale Smart Living (#93560)

gjohansson-ST committed:
Translate config flow for sensor measurement option in group (#87374)

* Translate config flow

* reset en

* Delete en.json

gjohansson-ST committed:
Add translation for Trafikverket Weatherstation sensor (#87315)

* Trafikverket weatherstation translate sensor

* Fix enums

freeDom- committed:
Glances: Add error handling for invalid sensor data (#93542)

* Set sensor to Unavailable
when native_value is invalid

* Add unit tests for sensors

ejpenney committed:
Obihai to have common unique_ids with DHCP or without (#91239)

* DHCPInfo and get_mac_address case mismatch

* Switch to format_mac

* Run black

starkillerOG committed:
Reolink autotrack streams for TrackMix and high/low resolution snapshots (#90591)

* Add auto track stream for Trackmix connected to NVR

* Enable autotracking stream by default

* Add snapshot high/low resolution

* Improve naming

* fix snapshots camera's not beeing added

* keep unique ID the same

* fix styling

bdraco committed:
Migrate restore_state helper to use registry loading pattern (#93773)

* Migrate restore_state helper to use registry loading pattern

As more entities have started using restore_state over time, it
has become a startup bottleneck as each entity being added is
creating a task to load restore state data that is already loaded
since it is a singleton

We now use the same pattern as the registry helpers

* fix refactoring error -- guess I am tired

* fixes

* fix tests

* fix more

* fix more

* fix zha tests

* fix zha tests

* comments

* fix error

* add missing coverage

* s/DATA_RESTORE_STATE_TASK/DATA_RESTORE_STATE/g

mkmer committed:
Calculate _attr_native_value when no restore state and appliance is running in Whirlpool (#88559)

* Initialize _attr_native_value when running

* Fix return type on update_sensor_state()
move init at startup if _attr_native_value is None

* allow update _attr_native_value when running and none

leranp committed:
Language codes for Hebrew (#93681)

* Language codes for Hebrew

There is 2 optional code for Hebrew:
he-IL is the new code
iw-IL is the old code , the google cloud STT for example is using the old code (iw)

* Update language.py

* Update test_language.py

* Update test_language.py

* Update test_language.py

* Simplify duplicate language check

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

raman325 committed:
Bump zwave-js-server-python to 0.49.0 (#93835)

* Bump zwave-js-server-python to 0.49.0

* fix tests

Lash-L committed:
Add non coordinated Entity to Roborock (#93841)

switch to non coordinator entity

rytilahti committed:
Swallow error 40000 for songpal power on/off (#80563)

* Swallow error 40000 for songpal power on/off

* Move ERROR_REQUEST_RETRY to consts

* Add tests for the swallow exception behavior

* Update tests/components/songpal/test_media_player.py

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Lash-L committed:
Rename confusing device info in Roborock (#93843)

rename confusing device info

raman325 committed:
Support zwave config parameters not on endpoint 0 (#93383)

* Support zwave config parameters not on endpoint 0

* Update device automation logic

* Make endpoint required

* Update homeassistant/components/zwave_js/services.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/services.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

jbouwh committed:
Remove MQTT platform schema warnings (#93785)

gjohansson-ST committed:
Add issue when integration key YAML not supported (#93807)

* Add issue when integration key YAML not supported

* Fix feedback

* Fix review

* Update homeassistant/components/homeassistant/strings.json

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

---------

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

emontnemery committed:
Drop incorrect CONFIG_SCHEMA from the homeassistant integration (#93850)

emontnemery committed:
Improve no-yaml warning (#93851)

emontnemery committed:
Add empty config schema to integrations t-z (#93853)

raman325 committed:
Cleanup unnecessary pylance warnings (#93849)

emontnemery committed:
Subscribe to device registry changes from entities (#93601)

* Subscribe to device registry changes from entities

* Use async_track_device_registry_updated_event

* Fix unsubscribe

* Fix logic, add tests

jbouwh committed:
Rework mqtt climate platform to simplify implementation water_heater platform (#93751)

* Rework to simplify water_heater platform

* Remove unused type hints

* Split shared and specific subscriptions

* Convert to abstract class

gjohansson-ST committed:
Fix Timer change service (#93469)

cdce8p committed:
Fix event typing (#93859)

raman325 committed:
Add support for Energy Production CC sensors (#93839)

raman325 committed:
Improve Z-Wave support for Ultraviolet multilevel sensor values (#93848)

Improve support for Ultraviolet multilevel sensor values

dmulcahey committed:
Bump ZHA quirks (#93860)

Noltari committed:
Use library constants for Airzone Cloud diagnostics (#93858)

airzone_cloud: diagnostics: use constants from library

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

yuxincs committed:
Add support for "days" unit for STESTI sensor in APCUPSD integration (#93844)

Add a test case for self test interval

Lash-L committed:
Add Status indicator light switch to Roborock (#93842)

add status indicator light switch

freeDom- committed:
Glances: Improve performance (#93852)

* Improve performance

* Apply suggestions from code review

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

Diegorro98 committed:
Matter cover position improvements (#92278)

mark007 committed:
Block legacy get-mac in preference of getmac (#87712)

Co-authored-by: GnatorX <garvinpang@protonmail.com>

c0ffeeca7 committed:
Standardize spelling of TTS and STT (#93857)

* Standardize spelling of TTS and STT

* Apply suggestions from code review

* Update homeassistant/components/tts/media_source.py

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

vanstinator committed:
Add time component to Melnor Bluetooth integration (#93652)

* Add time component to Melnor Bluetooth integration

* Apply suggestions from code review

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Simplify get pipeline method (#93865)

raman325 committed:
Add error handling for all zwave_js service calls (#93846)

* Add error handling for all service calls

* Switch siren to use internal function

* Remove failing checks

* Revert change to poll service, add comments, and add additional error handling

* Add error handling for ping and refresh + review comment + add tests

* Add test for statistics entity refresh

raman325 committed:
Give zwave_js platinum quality score (#93845)

farmio committed:
Update knx-frontend to 2023.5.31.141540 (#93864)

bramkragten committed:
Update frontend to 20230531.0 (#93862)

raman325 committed:
Move Z-Wave refresh value service out of task (#93866)

frenck committed:
Bump version to 2023.7.0dev0 (#93869)

tkdrob committed:
Add Dremel 3D Printer integration (#85969)

* Add Dremel 3D Printer integration

* remove validators requirement

* ruff

* uno mas

* uno mas

* uno mas

* uno mas

---------

Co-authored-by: Franck Nijhof <git@frenck.dev>
Co-authored-by: Tom Harris <tomharris@harrisnj.net>

balloob committed:
Pass device ID to conversation input (#93867)

jgrieger1 committed:
Add zwave_js speed config for additional GE/Jasco fan controllers (#92371)

* Add zwave_js speed config for additional GE/Jasco fan controllers

Add speed info for Honeywell(GE/Jasco) 39358 In-Wall Fan Speed Control and Enbrighten(GE/Jasco) 55258 In-Wall Fan Speed Control.

* Add test for Honeywell 39358 In-Wall Fan Speed Control

mdegat01 committed:
Refactor try catch in hassio.issues per feedback (#93872)

mkmer committed:
Address late review for whirlpool (#93874)

Address late review

tkdrob committed:
Add binary sensor platform to Dremel 3D Printer (#93881)

Add binary sensor platform to Dremel

puddly committed:
Increase Zigbee command retries (#93877)

* Enable retries for requests

* Update unit tests

* Account for fewer group retries in unit tests

Noltari committed:
Update aioairzone-cloud to v0.1.7 (#93871)

* Update aioairzone-cloud to v0.1.7

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* airzone_cloud: fix copy&paste description

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

---------

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

dgomes committed:
Always update Filter sensors attr on new_state  (#89096)

* always update attr

* reset filter on unit change

dgomes committed:
Delay filter integration until after HA has started (#91034)

* delay filter start

* Update homeassistant/components/filter/sensor.py

* Update homeassistant/components/filter/sensor.py

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Fix package names to match pypi index metadata (#93883)

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* Fix package names to match pypi index metadata

* uses _

* uses -

* fix metadata

balloob committed:
Bump frontend to 20230601.0 (#93884)

teharris1 committed:
Restructure Insteon start-up (#92818)

* Restructure startup

* Code review

* Further typing

* Fix circular import

emontnemery committed:
Add CONFIG_SCHEMA to broadlink (#93854)

* Add CONFIG_SCHEMA to broadlink

* Simplify error message

* Rename to no_yaml_config_schema

* Add tests

* Raise issue

* Tweak repairs issue description and title

* Update homeassistant/helpers/config_validation.py

* Add link

* Update homeassistant/components/homeassistant/strings.json

Co-authored-by: Franck Nijhof <git@frenck.dev>

* Fix logic, add test

---------

Co-authored-by: Franck Nijhof <git@frenck.dev>

emontnemery committed:
Remove async_setup from zerproc (#93903)

emontnemery committed:
Remove async_setup from ring (#93902)

emontnemery committed:
Remove setup from remote_rpi_gpio (#93901)

emontnemery committed:
Add config entry only config schema to integrations a-r (#93899)

joostlek committed:
Add video id to youtube sensor state attributes (#93668)

* Add video id to state attributes

* Make extra state attributes not optional

* Revert "Make extra state attributes not optional"

This reverts commit d2f9e936c809dd50a5e4bbdaa181c9c9ddd3d217.

emontnemery committed:
Add empty config schema to mobile_app (#93900)

emontnemery committed:
Remove async_setup from econet (#93892)

emontnemery committed:
Remove async_setup from gpslogger (#93893)

emontnemery committed:
Remove async_setup from hyperion (#93894)

emontnemery committed:
Add `silabs_multiprotocol` platform (#92904)

* Add silabs_multiprotocol platform

* Add new files

* Add ZHA tests

* Prevent ZHA from creating database during tests

* Add delay parameter to async_change_channel

* Add the updated dataset to the dataset store

* Allow MultipanProtocol.async_change_channel to return a task

* Notify user about the duration of migration

* Update tests

RenierM26 committed:
Add Ezviz light entity (#93710)

* Initial commit

* Add ezviz light entity.

* coveragerc

* Apply suggestions from code review

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

emontnemery committed:
Rename `cv.no_yaml_config_schema` to `cv.config_entry_only_config_schema` (#93908)

* Rename cv.no_yaml_config_schema to cv.config_entry_only_config_schema

* ✏️

emontnemery committed:
Remove async_setup from sky_hub (#93911)

emontnemery committed:
Remove async_setup from traccar (#93912)

dgomes reviewed (changes requested):
Currently there are several conflicts

disforw committed:
Adding myself as codeowner to QNAP (#93915)

* Adding codeowner and integration_type

* Update CODEOWNERS

* Update sorting

* Update integrations.json

emontnemery committed:
Fix typo in cloud (#93917)

disforw committed:
Move QNAP constants (#93918)

* Create const.py

* Update sensor.py

* Add docstring

* Update sensor.py

* Update homeassistant/components/qnap/sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update homeassistant/components/qnap/const.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

elupus committed:
Raise exception instead of hide in logs on zha write (#93571)

Raise exception instead of hide in logs

Write request that failed parsing of data would fail,
yet display as successful in the gui.

joostlek committed:
Bump python-opensky (#93916)

emontnemery committed:
Add platform only config schema to nsw_fuel_station and ping (#93913)

spyfly committed:
Fix LaMetric Config Flow for SKY (#93483)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

vingerha committed:
Properly order moon phases in attribute (#93933)

bdraco committed:
Fix onvif cameras that use basic auth with no password (#93928)

bdraco committed:
Bump python-onvif-zeep to 3.1.9 (#93930)

Xyaren committed:
Add humidity sensor to deconz component (#93024)

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

bdraco committed:
Bump pyunifiprotect to 4.9.1 (#93931)

bramkragten committed:
Update frontend to 20230601.1 (#93927)

andiukas committed:
Adding new supported language code to Google translate (#93926)

bdraco committed:
Fix typing_extensions to match metadata (#93920)

bdraco committed:
Make RestoreStateData.async_get_instance backwards compatible (#93924)

puddly committed:
Include port info in the ZHA websocket settings response (#93934)

emontnemery committed:
Add config entry only config schema to integrations s-z (#93910)

perdue committed:
Add save clips to Blink services (#84149)

emontnemery committed:
Remove async_setup from locative (#93895)

* Remove async_setup from locative

* Micro-optimize async_setup_entry

automaton82 committed:
Update netdata to 1.1.0, set longer timeout (#93937)

bdraco committed:
Fix august aiohttp session being closed out from under it (#93942)

* Fix august aiohttp session being closed out from under it

fixes #93941

* Fix august aiohttp session being closed out from under it

fixes #93941

* Fix august aiohttp session being closed out from under it

fixes #93941

bdraco committed:
Bump aiohomekit to 2.6.4 (#93943)

changelog: https://github.com/Jc2k/aiohomekit/compare/2.6.3...2.6.4

mostly additional logging to help track down #93891

tetele committed:
Fix states not being translated in voice assistants (#93572)

Fix states not being translated

ctalkington committed:
Update pyipp to 0.13.0 (#93886)

emontnemery committed:
Remove insteon import flow (#93952)

emontnemery committed:
Add empty config schema to mailbox (#93953)

emontnemery committed:
Add CONFIG_SCHEMA to imap_email_content (#93951)

* Add CONFIG_SCHEMA to imap_email_content

* Update homeassistant/components/imap_email_content/__init__.py

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

---------

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

emontnemery committed:
Add empty config schema to stt (#93954)

raman325 committed:
Make Z-Wave device IBT4ZWAVE discoverable as a cover (#93946)

* Make Z-Wave device IBT4ZWAVE discoverable as a cover

* Test device class

raman325 committed:
Improve logic for zwave_js.lock.is_locked attr (#93947)

dgomes committed:
Make Riemann sum sensors restore last valid state (#93674)

* keep last valid state

* keep last valid state

* typo

* increase coverage

* better error handling

* debug messages

* increase coverage

* remove random log

* don't expose last_valid_state as an attribute

chemelli74 committed:
Fritz: cleanup unused variables (#93971)

Cleanup

tkdrob committed:
Catch Google Sheets api error (#93979)

wrt54g committed:
Fix broken URL in Z-Wave JS (#93983)

Fix broken URL

gjohansson-ST committed:
Add scan interval to Command Line (#93752)

* Add scan interval

* Handle previous not complete

* Fix faulty text

* Add tests

* lingering

* Cool down

* Fix tests

tronikos committed:
Fix error in tibber while fetching latest statistics (#93998)

balloob committed:
Keep track what devices support Assist features (#93990)

Megabytemb committed:
Add reload support to intent_script (#93404)

* support live reload of intent_script

* add services.yaml

* update tesls for full code coverage

* Update based on feedback

* fix intent_script reload when no intent_script config

* Update homeassistant/helpers/intent.py

* update tests to handle no_existing better

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

gjohansson-ST committed:
Refactor Command Line binary sensor to use ManualTriggerEntity (#94000)

Refactor binary sensor ManualTriggerEntity

gjohansson-ST committed:
Refactor Command Line cover to use ManualTriggerEntity (#93997)

Refactor command_line cover

joostlek committed:
Show the sensor state using the coordinatordata instead of initial data (#94008)

* Show the sensor state using the coordinatordata instead of initial data

* Add test

* Remove part

tronikos committed:
Android TV Remote: Abort zeroconf if mac address is missing (#94026)

Abort zeroconf if mac address is missing

teaserrr committed:
Merge branch 'onvif-ha-event' of github.com:teaserrr/home-assistant into onvif-ha-event

MartinHjelmare commented:
Please start over in a fresh branch and PR. Thanks!

teaserrr commented:
> Please start over in a fresh branch and PR. Thanks!

Thanks, I just spent hours rebasing this.
